src/topology/algebra/floor_ring.lean

```diff
@@ -0,0 +1,225 @@
+/-
+Copyright (c) 2020 Anatole Dedecker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anatole Dedecker
+
+Basic topological facts (limits and continuity) about `floor`,
+`ceil` and `fract` in a `floor_ring`.
+-/
+import topology.algebra.ordered
+import algebra.floor
+
+open set function
+open_locale topological_space
+
+variables {α : Type*} [linear_ordered_ring α] [floor_ring α]
+
+lemma tendsto_floor_at_top :
+  filter.tendsto (floor : α → ℤ) filter.at_top filter.at_top :=
+begin
+  refine monotone.tendsto_at_top_at_top (λ a b hab, floor_mono hab) (λ b, _),
+  use (b:α)+((1:ℤ):α),
+  rw [floor_add_int, floor_coe],
+  exact (lt_add_one _).le
+end
+
+lemma tendsto_floor_at_bot :
+  filter.tendsto (floor : α → ℤ) filter.at_bot filter.at_bot :=
+begin
+  refine monotone.tendsto_at_bot_at_bot (λ a b hab, floor_mono hab) (λ b, ⟨b, _⟩),
+  rw floor_coe
```

Comment:
Should `exact_mod_cast` help you here? `floor_coe` sounds like the name of a `norm_cast` lemma.

Reply:
It doesn't but that's not surprising : `floor_coe` isn't really a coe-manipulation lemma, it just says the floor of (the coe of) an integer is itself. Maybe `floor_int` would be a better name ? (Like the already-existing `floor_add_int`)